docs/standard/design-guidelines/choosing-between-anonymous-and-tuple.md

```diff
@@ -0,0 +1,114 @@
+
+The previous C# snippet projects an anonymous type that would be defined with two properties, much like the following compiler-generated C# class:
+
+```csharp
+internal sealed class f__AnonymousType0
+{
+    public string Formatted { get; }
+    public long Ticks { get; }
+
+    public f__AnonymousType0(string formatted, long ticks)
+    {
+        Formatted = formatted;
+        Ticks = ticks;
+    }
+}
+```
+
+For more information, see [anonymous types](../../csharp/programming-guide/classes-and-structs/anonymous-types.md). The same functionality exists with tuples when projecting into LINQ queries, you can select properties and put them into tuples. These tuples flow through the query, just as anonymous types would. Now consider the following example using the `System.Tuple<string, long>`.
+
+```csharp-interactive
+var dates = new[]
+{
+    DateTime.UtcNow.AddHours(-1),
+    DateTime.UtcNow,
+    DateTime.UtcNow.AddHours(1),
+};
+
+foreach (var tuple in
+            dates.Select(
+                date => new Tuple<string, long>($"{date:MMM dd, yyyy hh:mm zzz}", date.Ticks)))
+{
+    Console.WriteLine($"Ticks: {tuple.Item2}, formatted: {tuple.Item1}");
+}
+```
+
+With the <xref:System.Tuple%602?displayProperty=nameWithType>, the instance exposes numbered item properties, such as `Item1`, and `Item2`. These property names can make it more difficult to understand the intent of the property values, as the property name only provides the ordinal. Furthermore, the `System.Tuple` types are reference `class` types. The <xref:System.ValueTuple%602?displayProperty=nameWithType> however, is a value `struct` type.
+
+```csharp-interactive
+var dates = new[]
+{
+    DateTime.UtcNow.AddHours(-1),
+    DateTime.UtcNow,
+    DateTime.UtcNow.AddHours(1),
+};
+
+foreach (var (formatted, ticks) in
+            dates.Select(
+                date => (Formatted: $"{date:MMM dd, yyyy at hh:mm zzz}", date.Ticks)))
+{
+    Console.WriteLine($"Ticks: {ticks}, formatted: {formatted}");
+}
+```
+
+The previous examples are all functionally equivalent, however; there are slight differences in their usability and their underlying implementations. One nicety with <xref:System.ValueTuple> is the ability to deconstruct.
+
+## Limitations
+
+While it might seem as though you should always use <xref:System.ValueTuple> over <xref:System.Tuple>, and anonymous types - there are some limitations that you should consider. The <xref:System.ValueTuple> types are mutable, whereas <xref:System.Tuple> are read-only. Anonymous types can be used in expression trees, while tuples cannot.
+
+### Key differences
+
+| Name                     | Type     | Custom property name | Deconstruction support | Expression tree support |
```

Comment:
Consider either adding a column, or note on the "Type" column that System.Tuple and System.ValueTuple are public, but anonymous types are internal to an assembly.

Another important tradeoff is that you can't use anonymous types as a parameter to method (other than as a generic type argument). That's because anonymous types don't have names.

Reply:
Anonymous types, System.Tuples and System.ValueTuples could be used with that function. However, anonymous types can't be arguments to non-generic methods.